File: examples/typescript/asset-example.ts
// Run: npx tsx asset-example.ts
// asset-example.ts -- demonstrates upload() and list() for file management
import { OneMinClient } from 'onemin';

const apiKey = process.env.ONEMIN_API_KEY;
if (!apiKey) {
  console.error('Set ONEMIN_API_KEY environment variable to run this example.');
  console.error('  export ONEMIN_API_KEY=your-key-here');
  process.exit(1);
}

const client = new OneMinClient({ apiKey });

// --- Demo 1: Upload a file (minimal 1x1 transparent PNG) ---
const tinyPng = new Uint8Array([
  0x89, 0x50, 0x4e, 0x47, 0x0d, 0x0a, 0x1a, 0x0a, 0x00, 0x00, 0x00, 0x0d,
  0x49, 0x48, 0x44, 0x52, 0x00, 0x00, 0x00, 0x01, 0x00, 0x00, 0x00, 0x01,
  0x08, 0x06, 0x00, 0x00, 0x00, 0x1f, 0x15, 0xc4, 0x89, 0x00, 0x00, 0x00,
  0x0a, 0x49, 0x44, 0x41, 0x54, 0x78, 0x9c, 0x63, 0x00, 0x01, 0x00, 0x00,
  0x05, 0x00, 0x01, 0x0d, 0x0a, 0x2d, 0xb4, 0x00, 0x00, 0x00, 0x00, 0x49,
  0x45, 0x4e, 0x44, 0xae, 0x42, 0x60, 0x82,
]);

const result = await client.asset.upload(['test.png', tinyPng]);
console.log('=== Asset Uploaded ===');
console.log(`Asset ID:     ${result.assetId}`);
console.log(`URL:          ${result.url}`);
console.log(`Content-type: ${result.contentType}`);
console.log();

// --- Demo 2: List assets ---
const assets = await client.asset.list();
console.log('=== Asset List ===');
if (Array.isArray(assets) && assets.length > 0) {
  for (const asset of assets.slice(0, 5)) {
    console.log(`  - ${asset.id} | ${asset.contentType} | ${asset.location ?? ''}`);
  }
} else {
  console.log('  (no assets found)');
}
console.log();

// --- Demo 3: Get a specific asset by ID ---
if (result.assetId) {
  const fetched = await client.asset.get(result.assetId);
  console.log('=== Asset Retrieved ===');
  console.log(`URL: ${fetched.url}`);
  console.log();
}

File: examples/python/asset_example.py
"""Asset domain example -- demonstrates upload() and list() for file management."""
import os
import sys
from onemin import OneMinClient

api_key = os.environ.get("ONEMIN_API_KEY")
if not api_key:
    print("Set ONEMIN_API_KEY environment variable to run this example.")
    print("  export ONEMIN_API_KEY=your-key-here")
    sys.exit(1)

client = OneMinClient(api_key=api_key)

# --- Demo 1: Upload a file (create a small test PNG in memory) ---
# Minimal 1x1 transparent PNG (67 bytes)
TINY_PNG = (
    b"\x89PNG\r\n\x1a\n\x00\x00\x00\rIHDR\x00\x00\x00\x01\x00\x00\x00\x01"
    b"\x08\x06\x00\x00\x00\x1f\x15\xc4\x89\x00\x00\x00\nIDATx\x9cc\x00\x01"
    b"\x00\x00\x05\x00\x01\r\n-\xb4\x00\x00\x00\x00IEND\xaeB`\x82"
)

result = client.asset.upload(("test.png", TINY_PNG))
print("=== Asset Uploaded ===")
print(f"Asset ID:     {result.asset_id}")
print(f"URL:          {result.url}")
print(f"Content-type: {result.content_type}")
print()

# --- Demo 2: List assets ---
assets = client.asset.list()
print("=== Asset List ===")
if assets:
    for asset in assets[:5]:  # show first 5
        print(f"  - {asset.get('id')} | {asset.get('contentType')} | {asset.get('location', '')}")
else:
    print("  (no assets found)")
print()

# --- Demo 3: Get a specific asset by ID ---
if result.asset_id:
    fetched = client.asset.get(result.asset_id)
    print("=== Asset Retrieved ===")
    print(f"URL: {fetched.url}")
    print()
